code/messaging/src/phone/java/com/adobe/marketing/mobile/Messaging.java

```diff
@@ -191,20 +204,116 @@ public static void refreshInAppMessages() {
 
     // region proposition retrieval api
 
-    // TODO: implement
+    /**
+     * Registers a permanent event listener with the Mobile Core for listening to personalization decisions events received upon a personalization query to the Experience Edge network.
+     * If a new {@code AdobeCallback} is provided, it will replace the existing one and it will be invoked when propositions are received from the Edge network.
+     *
+     * @param callback A {@link AdobeCallback} which will be invoked with a {@link Map<Surface, List<Proposition>>} containing the {@link Surface}s and the corresponding list of {@link Proposition} objects.
+     */
     public static void setPropositionsHandler(@NonNull final AdobeCallback<Map<Surface, List<Proposition>>> callback) {
```

Comment:
We plant to revisit this handler support because of its shortcomings. I've removed its support for the beta.